iroh-gateway/src/client.rs

```diff
@@ -23,8 +23,10 @@ impl Client {
         path: &str,
         rpc_client: &iroh_rpc_client::Client,
         start_time: std::time::Instant,
+        state: Arc<State>,
```

Comment:
- lets just store the `state` on the `Client`
- we don't need to pass `rpc_client` anymore, as that is stored in `state` as well

Reply:
sorry, I missed that the `client` is currently stored inside of `state`. so really, we should just unify them probably and implement this method on `State`, and do away with the `Client` struct

Reply:
Tbh I'm not entirely sold on combining all the client logic into the core gateway. I would rather abstract over client and push the rpc client into client itself and have:
- all the coordination logic in gateway
- all the fetching and resolution in client
- all the call execution in rpc client

Reply:
sure, that works